### PR TITLE
Add dependabot

### PR DESCRIPTION
**Pull request by lucacome**
**What this PR does / why we need it**:

Adds dependabot to keep dependencies up to date

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mengjiao-liu** [comment]
Add it to my to-do list.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version: 2
+updates:
+  - package-ecosystem: gomod
+    directories:
+      - "**/*"
+    schedule:
+      interval: daily
```
Comment (mengjiao-liu) [inline .github/dependabot.yml:7]:
```suggestion
      interval: "weekly"
```
Does the scheduling interval need to be set to daily? Is it too frequent? I think a weekly run is enough.

**Review by mengjiao-liu — COMMENTED** [review]
You may be more familiar with the dependabot configuration. Please double-check it. @pohly

File: .github/dependabot.yml (hunk 1 above)
Comment (lucacome) [reply .github/dependabot.yml:7]: I don't think it makes much of a difference, with weekly it's more likely that you'll get a bunch of PRs all at once and reach the limit of open PRs

Comment (pohly) [reply .github/dependabot.yml:7]: Which "limit of open PRs"?

klog does not have external Go dependencies. I am unsure what kind of PRs dependabot will create here and how useful they will be. I suppose we can try it out...

/lgtm
/assign @dims

For approval.

Comment (lucacome) [reply .github/dependabot.yml:7]: @pohly https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#open-pull-requests-limit

by default dependabot only opens 5 PRs for every package-ecosystem at a time

**Comment by lucacome** [comment]
Looks like dependabot is not enabled for the repo maybe? It should've opened a few PRs by now

**Comment by pohly** [comment]
PRs for what?

**Comment by lucacome** [comment]
GitHub Actions

**Pull request merged**